src/Service/WebHook/UpdateLineItemsService.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Service\WebHook;
+
+use _HumbugBoxf99c1794c57d\Symfony\Component\Console\Exception\LogicException;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\EntityManagerInterface;
+use OAT\SimpleRoster\Repository\LineItemRepository;
+use OAT\SimpleRoster\WebHook\UpdateLineItemCollection;
+use OAT\SimpleRoster\WebHook\UpdateLineItemDto;
+
+class UpdateLineItemsService
+{
+    /** @var LineItemRepository */
+    private $lineItemRepository;
+
+    /** @var EntityManagerInterface */
+    private $entityManager;
+
+    public function __construct(LineItemRepository $lineItemRepository, EntityManagerInterface $entityManager)
+    {
+        $this->lineItemRepository = $lineItemRepository;
+        $this->entityManager = $entityManager;
+    }
+
+    public function handleUpdates(UpdateLineItemCollection $collection): UpdateLineItemCollection
+    {
+        $knownUpdates = $collection
+            ->filter(
+                function (UpdateLineItemDto $dto): bool {
+                    return $dto->getName() === UpdateLineItemDto::NAME;
+                }
+            )
+            ->setStatus(UpdateLineItemDto::STATUS_ERROR);
+
+        $slugs = $knownUpdates
+            ->map(
+                function (UpdateLineItemDto $dto): string {
+                    return (string)$dto->getSlug();
+                }
+            );
+
+        if (count((array)$slugs) < 1) {
+            return $collection;
+        }
+
+        $lineItems = $this->lineItemRepository->findBy(
+            [
+                'slug' => (array)$slugs
+            ]
+        );
+
+        foreach ($lineItems as $lineItem) {
+            $duplicatedEvents = $knownUpdates
```

Comment:
IMO this confirms the `*Dto`should be renamed to `*Event`

Something like: `$duplicatedEvents = $knownUpdateEvents->filter()`

Reply:
I do not think that it's a good idea to bring these events to our bounded context. In our context, we have line items, not these events. if we rename it, it can cause confusion inside the project with the proper events that we can have into our application.